src/index.ts

```diff
@@ -13,29 +11,36 @@ const defaultPreLoadable: EnhancedPreLoadable = {
 
 let i = 0;
 
+const isCached = cache.has;
+
 const register = <P extends {}>(component: PreLoadable & Partial<EnhancedPreLoadable>) => {
+    const name = `Component${i++}`;
     const enhancedComponent: Component = {
-        name: `Component${i++}`,
+        name,
         ...defaultPreLoadable,
         ...component
     };
-    const { name } = enhancedComponent;
```

Comment:
This is incorrect pre-mature optimization...

The `name` can be overwritten in `component` variable 👀